pkg/ebpf/ebpftest/testlogwriter.go

```diff
@@ -0,0 +1,26 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package ebpftest
+
+import (
+	"testing"
+)
+
+// TestLogWriter wraps the testing.T object and provides a simple
+// io.Writer interface, to be used with DumpMaps functions
+// Very simple implementation now for output in debug functions so
+// newlines aren't handled: each call to Write is just sent to
+// t.Log
+type TestLogWriter struct {
+	T *testing.T
+}
+
+// Write method implementation, sends the data to t.Log()
+func (tlw *TestLogWriter) Write(p []byte) (int, error) {
+	tlw.T.Log(string(p))
```

Comment:
Did you find this useful? I could see this getting really noisy

Reply:
It's mostly to avoid having the pattern

```
var buffer bytes.Buffer
err := monitor.DumpMaps(&buffer, "name")

if err != nil {
  t.Logf("failed dumpMaps")
} else {
  t.Logf(buffer.String())
}
```

repeated all over the code. It's not a radical change, and from what I've seen in the output it doesn't seem to change things too much, but I can revert it if you prefer.